src/monitoring_tests/fee_monitoring/fee_monitoring.py

```diff
@@ -0,0 +1,239 @@
+            quote_response = requests.post(
+                prod_endpoint_url,
+                headers=header,
+                json=request_dict,
+                timeout=30,
+            )
+            if quote_response.status_code != SUCCESS_CODE:
+                barn_endpoint_url = "https://barn.api.cow.fi/mainnet/api/v1/quote"
+                quote_response = requests.post(
+                    barn_endpoint_url,
+                    headers=header,
+                    json=json.dumps(request_dict),
+                    timeout=30,
+                )
+                if quote_response.status_code != SUCCESS_CODE:
+                    TemplateTest.logger.error(
+                        "Quote error: %s.", quote_response.status_code
+                    )
+        except ValueError as except_err:
+            TemplateTest.logger.error("Unhandled exception: %s.", str(except_err))
+
+        quote_json = json.loads(quote_response.text)
+
+        quote_buy_amount = int(quote_json["quote"]["buyAmount"])
+        quote_sell_amount = int(quote_json["quote"]["sellAmount"])
+        quote_fee_amount = int(quote_json["quote"]["feeAmount"])
+
+        return quote_buy_amount, quote_sell_amount, quote_fee_amount
+
+    def get_solver_solution(  # pylint: disable=unused-argument
+        self,
+        decoded_settlement: DecodedSettlement,
+        i: int,
+    ) -> Tuple[int, int, int]:
+        """
+        Given a trade, compute buy_amount, sell_amount, and fee_amount of the trade
+        as proposed by a solver resolving the instance with only that order.
+        """
+        # solver_json = json.loads()
+
+        solver_buy_amount = 0
+        solver_sell_amount = 0
+        solver_fee_amount = 0
+
+        return solver_buy_amount, solver_sell_amount, solver_fee_amount
+
+    def fee_test(self, tx_hash) -> bool:  # pylint: disable=too-many-locals
+        """
+        Given a transaction hash, check if there is a partially-fillable order in the settlement.
+        If this is the case, perform multiple tests on the execution of those orders to check if
+        the fee was set correctly.
+        TODO: add more explanations
+        """
+        # get trades via api
+        orders = TemplateTest.get_endpoint_order_data(tx_hash)
+        # loop through trades
+
+        encoded_transaction = TemplateTest.get_encoded_transaction(tx_hash)
+        decoded_settlement = TemplateTest.get_decoded_settlement_raw(
+            encoded_transaction
+        )
+
+        partially_fillable_indices = []
+        for i, order in enumerate(orders):
+            if order["partiallyFillable"]:
+                partially_fillable_indices.append(i)
+
+        if len(partially_fillable_indices) > 0:
+            # get additional data for the batch
+            receipt = TemplateTest.get_encoded_receipt(tx_hash)
+            competition_data = TemplateTest.get_solver_competition_data([tx_hash])[0]
```

Comment:
this line currently errors (if competition data is not ready yet). Returning `False` is what the endpoint test does.
```suggestion
            competition_data_list = TemplateTest.get_solver_competition_data([tx_hash])
            if len(competition_data_list) == 0:
                return False
            competition_data = competition_data_list[0]
```